File: app/src/main/java/ado/com/testarchcomponents/view/StartActivity.java
package ado.com.testarchcomponents.view;

import android.arch.lifecycle.Observer;
import android.arch.lifecycle.ViewModelProviders;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;

import ado.com.testarchcomponents.model.GpsLocation;
import ado.com.testarchcomponents.viewmodel.LocationViewModel;
import ado.com.testarchcomponents.R;
import ado.com.testarchcomponents.databinding.ActivityMainBinding;

public class StartActivity extends AppCompatActivity implements Observer<GpsLocation> {
    private LocationViewModel mViewModel;
    private ActivityMainBinding mBinding;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        mViewModel = ViewModelProviders.of(this).get(LocationViewModel.class);
        //instead of pushing data to the UI, let the UI observe changes to it.
        mViewModel.getLocationLiveData().observe(this, this);
    }

    @Override
    public void onDestroy() {
        super.onDestroy();
        mViewModel.getLocationLiveData().removeObserver(this);
    }

    @Override
    public void onChanged(@Nullable GpsLocation location) {
        if (location != null) {
            mBinding.included.setLocation(location);
        }
    }
}

File: app/src/main/java/ado/com/testarchcomponents/view/StartActivity.kt
package ado.com.testarchcomponents.view

import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v7.app.AppCompatActivity

import ado.com.testarchcomponents.model.GpsLocation
import ado.com.testarchcomponents.viewmodel.LocationViewModel
import ado.com.testarchcomponents.R
import ado.com.testarchcomponents.databinding.ActivityMainBinding

class StartActivity : AppCompatActivity(), Observer<GpsLocation> {
    private var mViewModel: LocationViewModel? = null
    private var mBinding: ActivityMainBinding? = null

    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        setupList()
        mViewModel = ViewModelProviders.of(this).get(LocationViewModel::class.java)
        //instead of pushing data to the UI, let the UI observe changes to it.
        mViewModel?.locationLiveData?.observe(this, this)
    }

    private fun setupList() {
        mBinding?.included?.coordinatesList?.adapter = CoordinatesAdapter()
        subscribeToListChanges()
    }

    private fun subscribeToListChanges() {
        //TODO subscribe for changes to the list. use a view-model which will observe the db. (try use room)
    }

    override fun onChanged(location: GpsLocation?) {
        if (location != null) {
            mBinding?.included?.location = location
        }
    }
}
